src/Api/SearchApiBase.php

```diff
@@ -117,7 +114,7 @@ public function getQuery(): ?array
     /**
      * @inheritDoc
      */
-    public function setSort(?array $sort): SearchApiBaseInterface
+    public function setSort($sort): SearchApiBaseInterface
```

Comment:
Yeah, this proves that they are different and they should not share the same interface. We can even name them differently to make it clear, e.g. `setFacetSort()`?